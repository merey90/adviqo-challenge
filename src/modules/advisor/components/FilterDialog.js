import React from 'react';
import {
  Dialog,
  DialogActions,
  DialogTitle,
  DialogContent,
  FormControl,
  FormLabel,
  FormControlLabel,
  RadioGroup,
  Radio,
  Select,
  MenuItem,
  Button
} from '@material-ui/core';
import SearchIcon from '@material-ui/icons/Search';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  form: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItems: 'center'
  },
  formControl: {
    display: 'flex'
  }
});

function FilterDialog(props) {
  const {
    isFiltering,
    filter,
    handleFilterSubmit,
    handleInputChange,
    triggerFilterDialog,
    classes
  } = props;
  const languages = ['all', 'english', 'kazakh'];
  return (
    <Dialog
      open={isFiltering}
      onClose={triggerFilterDialog}
      aria-labelledby="form-dialog-title"
      >
      <DialogTitle id="form-dialog-title">Filter</DialogTitle>
      <DialogContent>
      <FormControl >
        <FormLabel>Status</FormLabel>
          <RadioGroup
            row
            name="status"
            aria-label="color"
            value={filter.status}
            onChange={handleInputChange}
          >
            <FormControlLabel value="all" control={<Radio />} label="All" />
            <FormControlLabel value="online" control={<Radio />} label="Online" />
            <FormControlLabel value="offline" control={<Radio />} label="Offline" />
          </RadioGroup>
        </FormControl>
        <FormControl className={classes.formControl}>
          <FormLabel>Language</FormLabel>
          <Select
            value={filter.language}
            onChange={handleInputChange}
            inputProps={{
              name: 'language',
              id: 'language-simple',
            }}
          >
            {languages.map((language, index) =>
              (<MenuItem key={index} value={language}>
              {language.charAt(0).toUpperCase() + language.slice(1)}
              </MenuItem>))}
          </Select>
        </FormControl>
      </DialogContent>
      <DialogActions>
        <Button variant="contained" color="secondary"
          onClick={handleFilterSubmit}
        >
          Filter
          <SearchIcon />
        </Button>
      </DialogActions>
    </Dialog>
  )
}

export default withStyles(styles)(FilterDialog);