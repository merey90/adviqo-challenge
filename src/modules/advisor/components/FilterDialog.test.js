import React from 'react';
import { shallow } from 'enzyme';

import FilterDialog from './FilterDialog';

const filter = {
  language: 'all',
  status: 'all',
  sortBy: 'reviews',
  sortDirection: 'desc'
};

test('render FilterDialog', () => {
  const wrapper = shallow(
    <FilterDialog
      isFiltering
      filter={filter}
      handleInputChange={()=>{}}
      handleFilterSubmit={()=>{}}
      triggerFilterDialog={()=>{}}/>
  );
  expect(wrapper).toMatchSnapshot();
});