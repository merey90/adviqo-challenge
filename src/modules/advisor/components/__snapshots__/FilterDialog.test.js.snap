// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FilterDialog 1`] = `
<FilterDialog
  classes={
    Object {
      "form": "FilterDialog-form-1",
      "formControl": "FilterDialog-formControl-2",
    }
  }
  filter={
    Object {
      "language": "all",
      "sortBy": "reviews",
      "sortDirection": "desc",
      "status": "all",
    }
  }
  handleFilterSubmit={[Function]}
  handleInputChange={[Function]}
  isFiltering={true}
  triggerFilterDialog={[Function]}
/>
`;
